Sources/PackageGraph/Pubgrub/PubgrubDependencyResolver.swift

```diff
@@ -1410,10 +1416,10 @@ fileprivate extension PackageRequirement {
 
 fileprivate extension DependencyResolutionNode {
     var nameForDiagnostics: String {
-        if let product = specificProduct {
-            return "\(package.name)[\(product)]"
+        if let product = specificProduct, product != package.name{
```

Comment:
I like the simplification of not showing the product name when it is the same as the package name, but could it lead to confusion if some products are shown and others not?  It will not be possible to distinguish the "no-product" case visually from the case where the product has the same name as the package.

Perhaps this isn't actually a problem in practice, and the simplification certainly makes things read better.  I think that the product name started being listed there as part of the changes for target-based dependencies.

Reply:
good point @abertelrud. been thinking about this more and I now suspect we can drop the product name in all cases, including when the package and product names are different (which is the common case). since the error message explains why specific combination of package versions can't work together, I am not convinced the product names are material to understanding the issue - its more of additional/supplemental information resulting from target based deps - and as such I dont think worth the clutter. I proposed we remove it in all cases.

Reply:
I agree.  I'm not sure of the reasoning for why it was added, but since the dependencies are between packages, I agree it can be removed.